pkg/autoops/api/api.go

```diff
@@ -632,6 +632,55 @@ func (s *AutoOpsService) UpdateAutoOpsRule(
 		if err != nil {
 			return err
 		}
+
+		if autoOpsRule.AlreadyTriggered() || autoOpsRule.IsFinished() || autoOpsRule.IsStopped() {
+			dt, err := statusAutoOpsRuleCompleted.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalize(locale.InvalidArgumentError),
+			})
+			if err != nil {
+				return statusInternal.Err()
+			}
+			return dt.Err()
+		}
+		if autoOpsRule.OpsType == autoopsproto.OpsType_SCHEDULE &&
+			len(req.AddOpsEventRateClauseCommands) > 0 || len(req.ChangeOpsEventRateClauseCommands) > 0 {
+			dt, err := statusIncompatibleOpsType.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "ops_type"),
+			})
+			if err != nil {
+				return statusInternal.Err()
+			}
+			return dt.Err()
+		}
+		if autoOpsRule.OpsType == autoopsproto.OpsType_EVENT_RATE &&
+			len(req.AddDatetimeClauseCommands) > 0 || len(req.ChangeDatetimeClauseCommands) > 0 {
+			dt, err := statusIncompatibleOpsType.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "ops_type"),
+			})
+			if err != nil {
+				return statusInternal.Err()
+			}
+			return dt.Err()
+		}
+
+		// Changes to a different opsType are not allowed
+		if req.ChangeAutoOpsRuleOpsTypeCommand != nil &&
+			req.ChangeAutoOpsRuleOpsTypeCommand.OpsType != autoOpsRule.OpsType {
```

Comment:
Should it be the following?
`ChangeAutoOpsRuleOpsTypeCommand` is deprecated, so the command should not be used anymore, shouldn't it?

```suggestion
		if req.ChangeAutoOpsRuleOpsTypeCommand != nil {
```

Reply:
@kentakozuka 
Currently, it seems that this command is sent even if the OpsType is not changed when updating from the management screen.
https://github.com/bucketeer-io/bucketeer/blob/cc1466b8607a15323c6d06a829ef77dc2d4ebd8c/ui/web-v2/src/components/OperationAddUpdateForm/index.tsx#L255
Therefore, if you fix it, AutoOps will probably not be able to update.
How about not making the correction in this PR, but making the correction after the correction is made on the management screen?